### PR TITLE
Entity Interpolation

### PR DESCRIPTION
**Pull request by tvillegas98**
This PR introduces the concept of entity interpolation.

## How does it work?
Now we're buffering the last 30 updates received from the server (Created a new class called `EventsBuffer` to address this), and these updates comes with the server timestamp. That timestamp help us to render at least 3 ticks in the past. 
The main thing is that entity interpolation is applied to other entities, e.g, other players besides us.

## Disclaimers
- You'll see a new button in the backend playground called "Interpolation On/off.
- By default, interpolation and prediction are on. 
By the things mentioned above, BackendPlayground is present in the files changed.

## Todo Tasks
- [x] Reallocate the events buffer
- [x] Add toggle button to turn off the interpolation
- [ ] Draw the interpolation's ghost (it'll addressed in another PR)
- [x] Fix teleport
- [ ] Add a parameter for the amount of ticks to rollback (it'll addressed in another PR)
- [x] Fix Index errors

### CODE REVIEW

**Review by jrchatruc — APPROVED** [review]

File: server/lib/dark_worlds_server/engine/runner.ex [hunk 1]
```diff
@@ -296,7 +296,6 @@ defmodule DarkWorldsServer.Engine.Runner do
     {:ok, game} = create_new_game(opts.game_config, gen_server_state.max_players, selected_players)
 
     Logger.info("#{DateTime.utc_now()} Starting runner, pid: #{inspect(self())}")
-
```
Comment (manucamejo) [inline server/lib/dark_worlds_server/engine/runner.ex:299]: Restore this empty line

File: client/Assets/Scripts/SocketConnectionManager.cs [hunk 2]
```diff
@@ -161,11 +175,21 @@ private void OnWebSocketMessage(byte[] data)
...
+                        game_event.SelectedCharacters
+                    );
                     break;
                 case GameEventType.FinishCharacterSelection:
-                    this.selectedCharacters = fromMapFieldToDictionary(game_event.SelectedCharacters);
+                    this.selectedCharacters = fromMapFieldToDictionary(
+                        game_event.SelectedCharacters
+                    );
                     this.gamePlayers = game_event.Players.ToList();
+                    this.gamePlayers.ForEach(
```
Comment (manucamejo) [inline client/Assets/Scripts/SocketConnectionManager.cs]: remove this

File: client/Assets/Scripts/PlayerMovement.cs [hunk 3]
```diff
@@ -104,22 +120,54 @@ public void SendPlayerMovement()
...
-        GameEvent gameEvent = SocketConnectionManager.Instance.gameEvent;
+        long auxAccumulatedTime;
+        long currentTime;
+        long pastTime;
+        EventsBuffer buffer = SocketConnectionManager.Instance.eventsBuffer;
+        GameEvent gameEvent;
         for (int i = 0; i < SocketConnectionManager.Instance.gamePlayers.Count; i++)
         {
+            // We don't need to interpolate what we're seeing, just what we see about other players
+            if (buffer.firstTimestamp == 0)
```
Comment (manucamejo) [inline client/Assets/Scripts/PlayerMovement.cs:131]: Can you do this before the for statement?

File: server/lib/dark_worlds_server/engine/runner.ex (hunk 1 above)
Comment (tvillegas98) [reply server/lib/dark_worlds_server/engine/runner.ex]: done!

File: client/Assets/Scripts/SocketConnectionManager.cs (hunk 2 above)
Comment (tvillegas98) [reply client/Assets/Scripts/SocketConnectionManager.cs]: done!

File: client/Assets/Scripts/PlayerMovement.cs (hunk 3 above)
Comment (tvillegas98) [reply client/Assets/Scripts/PlayerMovement.cs]: done!

**Review by manucamejo — APPROVED** [review]

**Pull request merged**